pkg/network/ebpf/c/protocols/kafka/helpers.h

```diff
@@ -0,0 +1,287 @@
+        return false;
+    }
+
+    READ_BIG_ENDIAN_WRAPPER(s32, timeout_ms, skb, offset);
+    if (timeout_ms < 0) {
+        // timeout_ms cannot be negative.
+        return false;
+    }
+
+    *out_offset = offset;
+    return true;
+}
+
+// Getting the offset the topic name in the fetch request.
+static __always_inline u32 get_topic_offset_from_fetch_request(const kafka_header_t *kafka_header, struct __sk_buff *skb) {
+    // replica_id => INT32
+    // max_wait_ms => INT32
+    // min_bytes => INT32
+    u32 offset = 3 * sizeof(s32);
+
+    if (kafka_header->api_version >= 3) {
+        // max_bytes => INT32
+        offset += sizeof(s32);
+        if (kafka_header->api_version >= 4) {
+            // isolation_level => INT8
+            offset += sizeof(s8);
+            if (kafka_header->api_version >= 7) {
+                // session_id => INT32
+                // session_epoch => INT32
+                offset += 2 * sizeof(s32);
+            }
+        }
+    }
+
+    return offset;
+}
+
+// Calls the relevant function, according to the api_key.
+static __always_inline bool is_kafka_request(const kafka_header_t *kafka_header, struct __sk_buff *skb, u32 offset) {
+    // Due to old-verifiers limitations, if the request is fetch or produce, we are calculating the offset of the topic
+    // name in the request, and then validate the topic. We have to have shared call for validate_first_topic_name
+    // as the function is huge, rather than call validate_first_topic_name for each api_key.
+    switch (kafka_header->api_key) {
+    case KAFKA_PRODUCE:
+        if (!get_topic_offset_from_produce_request(kafka_header, skb, &offset)) {
+            return false;
+        }
+        break;
+    case KAFKA_FETCH:
+        offset += get_topic_offset_from_fetch_request(kafka_header, skb);
+        break;
+    default:
+        return false;
+    }
+    return validate_first_topic_name(skb, offset);
+}
+
+// Checks if the packet represents kafka communication.
+static __always_inline bool is_kafka(struct __sk_buff *skb, skb_info_t *skb_info, const char* buf, __u32 buf_size) {
+    CHECK_PRELIMINARY_BUFFER_CONDITIONS(buf, buf_size, KAFKA_MIN_LENGTH);
+
+    const kafka_header_t *header_view = (kafka_header_t *)buf;
+    kafka_header_t kafka_header;
+    bpf_memset(&kafka_header, 0, sizeof(kafka_header));
+    kafka_header.message_size = bpf_ntohl(header_view->message_size);
+    kafka_header.api_key = bpf_ntohs(header_view->api_key);
+    kafka_header.api_version = bpf_ntohs(header_view->api_version);
+    kafka_header.correlation_id = bpf_ntohl(header_view->correlation_id);
+    kafka_header.client_id_size = bpf_ntohs(header_view->client_id_size);
+
+    if (!is_valid_kafka_request_header(&kafka_header)) {
+        return false;
+    }
+
+    u32 offset = skb_info->data_off + sizeof(kafka_header_t);
+    // Validate client ID
+    // Client ID size can -1 if the client id is null.
+    if (kafka_header.client_id_size > 0) {
```

Comment:
What about validating it's not less than -1?

Reply:
added